packages/ui/src/utils/utilsFormat.ts

```diff
@@ -55,169 +33,25 @@ export function getFractionDigitsOptions(val: number | string | undefined | null
-      }
-
-      if (Number(val) === 0) {
-        parsedOptions.trailingZeroDisplay = 'stripIfInteger'
-        return new Intl.NumberFormat(localeDetected, parsedOptions).format(Number(val))
-      } else {
-        const numVal = Number(val)
-        const parsedVal = new BigNumber(val).toString()
-
-        if (shouldUseDefaultFractionDigits(parsedOptions)) {
-          if (showAllFractionDigits) {
-            parsedOptions.minimumFractionDigits = new BigNumber(Number(val)).toString().split('.')[1]?.length ?? 0
-          } else if (showDecimalIfSmallNumberOnly && numVal > 10) {
-            parsedOptions.maximumFractionDigits = 0
-          } else if (options?.style === 'percent') {
-            parsedOptions.maximumFractionDigits = getDecimal(parsedVal, 2)
-          } else if (notation === 'compact') {
-            delete parsedOptions.minimumFractionDigits
-            delete parsedOptions.maximumFractionDigits
-            if (numVal >= 1000000) {
-              parsedOptions.maximumFractionDigits = getDecimal(parsedVal, 2)
-              parsedOptions.notation = notation
-            } else {
-              if (numVal > 10) {
-                parsedOptions.trailingZeroDisplay = 'stripIfInteger'
-                parsedOptions.maximumFractionDigits = 0
-                parsedOptions.minimumFractionDigits = 0
-              } else {
-                parsedOptions.minimumFractionDigits = getDecimal(parsedVal, 5)
-              }
-            }
-          } else {
-            parsedOptions.minimumFractionDigits = getDecimal(parsedVal, 5)
-          }
-        }
-
-        if (Number(val) < 1) {
-          // reformat formatted number showing 0, but !== 0
-          if (formattedNumberIsZero(val, parsedOptions)) {
-            if (Number(val) < 0.000000001 && parsedOptions.style !== 'percent') {
-              return '<0.000000001'
-            } else {
-              const { minimumFractionDigits, maximumFractionDigits, ...rest } = parsedOptions
-              return _formatNumber(val, { ...rest, minimumSignificantDigits: 2, maximumSignificantDigits: 2 })
-            }
-          } else if (Number(val) <= 0.0009 && !('showAllFractionDigits' in (options ?? {}))) {
-            const { minimumFractionDigits, maximumFractionDigits, ...rest } = parsedOptions
-            // format number to maximumSignificantDigits of 4 if value is <= 0.0009
-            return _formatNumber(val, { ...(rest ?? {}), maximumSignificantDigits: 4 })
-          }
-        }
-
-        return _formatNumber(val, { ...parsedOptions, ...numberFormatOptions })
-      }
-    }
-  } catch (error) {
-    console.error(error, { params: { val, options } })
-    if (typeof val === 'undefined' || val === null || val === '') {
-      return defaultValue || ''
-    } else if (lodash.isNaN(Number(val))) {
-      return '?'
-    } else {
-      return val.toString()
-    }
-  }
+  if (val === undefined) return options?.defaultValue ?? '-'
```

Comment:
The `-` should be moved to the destruct statement imo
We should use `==` so it also catches `null`
```suggestion
  if (val == null) return defaultValue ?? '-'
```